django/contrib/postgres/search.py

```diff
@@ -92,14 +92,38 @@ def __init__(self, lhs, connector, rhs, config, output_field=None):
         super(CombinedSearchVector, self).__init__(lhs, connector, rhs, output_field)
 
 
-class SearchQuery(Value):
+class SearchQueryCombinable(object):
+    BITAND = '&&'
+    BITOR = '||'
+
+    def _combine(self, other, connector, reversed, node=None):
+        if not isinstance(other, SearchQueryCombinable) or not self.config == other.config:
+            raise TypeError('SearchQuery can only be combined with other SearchQuerys')
+        if reversed:
+            return CombinedSearchQuery(other, connector, self, self.config)
+        return CombinedSearchQuery(self, connector, other, self.config)
+
+    # On Combinable, these are not implemented to reduce confusion with Q. In
+    # this case we are actually (ab)using them to do logical combination so
+    # it's consistent with other usage in Django.
+    def __or__(self, other):
+        return self._combine(other, self.BITOR, False)
+
+    def __ror__(self, other):
```

Comment:
I don't see any tests failing if the ror and rand methods are removed. I guess that's a deficiency of the current test suite. Have you learned enough to add tests though (in a separate commit, probably, since it's not related to the other issue that fixed).